### ISSUE TITLE
message:"gcloud is installed but cannot read its version"

### ISSUE THREAD

**Issue by svenjeppsson** [GoogleCloudPlatform/cloud-code-vscode#112]
**Environment**:
Extension version: 0.0.8
VSCode version: Version: 1.34.0
Chrome: 66.0.3359.181
Google Cloud SDK 248.0.0
app-engine-go 
app-engine-java 1.9.74
app-engine-python 1.9.85
app-engine-python-extras 1.9.74
bq 2.0.43
cloud-datastore-emulator 2.1.0
core 2019.05.24
gsutil 4.38
kubectl 2019.05.24

OS: Linux x64 4.19.0-5-amd64
Distributor ID:	Debian
Description:	Debian GNU/Linux 10 (buster)
Release:	10
Codename:	buster
***

**Description / Repo steps**:
I fresh installed gcloud and the cloude-code-vscode plugin. I used the gcloud .tgz to install it. With my first installation via apt-get I came to the same problems.

When I open the cloud-code view in vscode an message: "gcloud is installed but cannot read its version" pops up and the "google kubernetes engine explorer" shows: "click here to log-in....". 

When I *"click here to log-in...."* click on it, the console opens the gcloud init dialog, I choose my already existing user and project. gcloud init runs without any error. after that ,  "google kubernetes engine explorer" still shows "click here to log-in....".

I tried create an GKE Cluster, there I cannot choose the the project-id or the zone.

**Comment by redborian** [comment]
Thanks for reporting the issue.
Can you run `gcloud info --format=json` and see if you can get the version of gcloud?

**Comment by rustyx** [comment]
Open Terminal inside VSCode (**Ctrl+\`**), in there enter `gcloud version`. You'll most likely see an error about a wrong Python version or another path problem. Fix the problem, restart VSCode.

**Comment by Sohojoe** [comment]
I have the same problem - I get this when I type: gcloud version from the terminal inside VS Code

```
Google Cloud SDK 251.0.0
bq 2.0.43
core 2019.06.14
gsutil 4.38
```

**Comment by cheung-chifung** [comment]
Same problem
```
Google Cloud SDK 252.0.0
bq 2.0.43
core 2019.06.21
gsutil 4.39
```

**Comment by redborian** [comment]
@cheung-chifung Can you run `gcloud info --format=json` and see if you can get the version of gcloud?

**Comment by cheung-chifung** [comment]
@redborian 

I ran `gcloud info --format=json` and got a Python value error `ERROR: gcloud crashed (ValueError): unknown locale: UTF-8`.
I googled it and put the following lines into my `~/.zshrc`

```
export LC_ALL=en_US.UTF-8
export LANG=en_US.UTF-8
```

It fixed the Value Error issue, `gcloud info --format=json` works well now, in both iTerm2 and VSCode terminal.

However, the `cannot read its version` problem is still there, even I rebooted my computer.

Here is my gcloud info with some private fields replaced.

```
{
  "basic": {
    "architecture": {
      "file_name": "x86_64",
      "id": "x86_64",
      "name": "x86_64"
    },
    "locale": [
      "en_US",
      "UTF-8"
    ],
    "operating_system": {
      "file_name": "darwin",
      "id": "MACOSX",
      "name": "Mac OS X"
    },
    "python_location": "/Users/cheung/.pyenv/versions/2.7.9/bin/python2",
    "python_version": "2.7.9 (default, Dec 11 2016, 14:47:12) \n[GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.42.1)]",
    "site_packages": false,
    "version": "252.0.0"
  },
  "config": {
    "account": "xxx@gmail.com",
    "active_config_name": "cheung-gke",
    "paths": {
      "active_config_path": "/Users/cheung/.config/gcloud/configurations/config_cheung-gke",
      "global_config_dir": "/Users/cheung/.config/gcloud",
      "installation_properties_path": "/usr/local/bin/google-cloud-sdk/properties"
    },
    "project": "xxxxx",
    "properties": {
      "core": {
        "account": "xxx@gmail.com",
        "disable_usage_reporting": "False",
        "project": "xxx"
      }
    }
  },
  "env_proxy": {
    "address": null,
    "password": null,
    "port": null,
    "type": null,
    "username": null
  },
  "installation": {
    "additional_repos": [],
    "components": {
      "bq": "2.0.43",
      "core": "2019.06.21",
      "gsutil": "4.39"
    },
    "duplicate_tool_paths": [],
    "kubectl": "/usr/local/bin/kubectl",
    "old_tool_paths": [],
    "on_path": true,
    "path": [
      "/Users/cheung/.pyenv/versions/2.7.9/bin",
      "/usr/local/Cellar/pyenv/1.1.5/libexec",
      "/Users/cheung/.miniconda3/bin",
      "/usr/local/Cellar/pyenv-virtualenv/1.1.1/shims",
      "/Users/cheung/.pyenv/shims",
      "/usr/local/bin/google-cloud-sdk/bin",
      "/usr/local/Cellar/pyenv-virtualenv/1.1.1/shims",
      "/Users/cheung/.pyenv/shims",
      "/Users/cheung/.opam/system/bin",
      "/usr/local/opt/go/libexec/bin",
      "/Users/cheung/Projects/gocode/bin",
      "/Users/cheung/.cargo/bin",
      "/usr/local/bin",
      "/usr/bin",
      "/bin",
      "/usr/sbin",
      "/sbin",
      "/usr/local/share/dotnet",
      "/Library/Frameworks/Mono.framework/Versions/Current/Commands",
      "/Users/cheung/.miniconda3/bin",
      "/usr/local/Cellar/pyenv-virtualenv/1.1.1/shims",
      "/Users/cheung/.pyenv/shims",
      "/usr/local/bin/google-cloud-sdk/bin",
      "/Users/cheung/.opam/system/bin",
      "/usr/local/opt/go/libexec/bin",
      "/Users/cheung/Projects/gocode/bin",
      "/Users/cheung/.cargo/bin",
      "/Users/cheung/.fzf/bin"
    ],
    "python_path": [
      "/usr/local/bin/google-cloud-sdk/lib/third_party",
      "/usr/local/bin/google-cloud-sdk/lib",
      "/Users/cheung/.pyenv/versions/2.7.9/lib/python27.zip",
      "/Users/cheung/.pyenv/versions/2.7.9/lib/python2.7/",
      "/Users/cheung/.pyenv/versions/2.7.9/lib/python2.7/plat-darwin",
      "/Users/cheung/.pyenv/versions/2.7.9/lib/python2.7/plat-mac",
      "/Users/cheung/.pyenv/versions/2.7.9/lib/python2.7/plat-mac/lib-scriptpackages",
      "/Users/cheung/.pyenv/versions/2.7.9/lib/python2.7/lib-tk",
      "/Users/cheung/.pyenv/versions/2.7.9/lib/python2.7/lib-old",
      "/Users/cheung/.pyenv/versions/2.7.9/lib/python2.7/lib-dynload"
    ],
    "release_channel": "rapid",
    "repo_url": "https://dl.google.com/dl/cloudsdk/channels/rapid/components-2.json",
    "sdk_root": "/usr/local/bin/google-cloud-sdk"
  },
  "logs": {
    "last_log": "/Users/cheung/.config/gcloud/logs/2019.06.29/08.59.41.088714.log",
    "last_logs": [
      "/Users/cheung/.config/gcloud/logs/2019.06.29/08.59.41.088714.log",
      "/Users/cheung/.config/gcloud/logs/2019.06.29/08.59.28.235032.log",
      "/Users/cheung/.config/gcloud/logs/2019.06.29/08.59.27.737620.log",
      "/Users/cheung/.config/gcloud/logs/2019.06.29/08.59.27.228720.log",
      "/Users/cheung/.config/gcloud/logs/2019.06.29/08.59.26.744411.log"
    ],
    "logs_dir": "/Users/cheung/.config/gcloud/logs"
  },
  "tools": {
    "git_version": "git version 2.15.1",
    "ssh_version": "OpenSSH_7.9p1, LibreSSL 2.7.3"
  }
}
```

**Comment by redborian** [comment]
@cheung-chifung Your VSCode instance would also need to have those environment variables set. Can you start VSCode from terminal following these steps (https://stackoverflow.com/questions/30065227/run-open-vscode-from-mac-terminal) ?

**Comment by cheung-chifung** [comment]
Thank you, I tried it with no luck.

**Comment by redborian** [comment]
We have fixed the underlying issue in the next release which is due soon. I will update this thread once it is released.

**Comment by sujit-kamireddy** [comment]
This issue has been fixed in the latest release 0.0.10. Please try it out.

As always, please keep issues coming our way.
